### ISSUE TITLE
Tray with only title

### ISSUE THREAD

**Issue by vikeri** [electron/electron#5439]
- Electron version: 0.37.8
- Operating system: OSX

Is there an option to have a tray without an icon on OSX? I wan't to show only text namely, similar to how [bitbar](https://github.com/matryer/bitbar) works. If I set the icon to a 1px transparent PNG (the hacky way), I get this result (see whitespace to the left):

![image](https://cloud.githubusercontent.com/assets/846688/15093354/f4c5c12a-1485-11e6-8dd3-859e9c868116.png)

To clarify, I'm doing an OSX only app at the moment.

**Comment by bengotow** [comment]
Hey! I don't think there's a way to do one without an icon currently. Definitely should be added, but in the interim, you might be able to pre-render images for [0-100%] and use images instead of the title.

**Comment by vikeri** [comment]
Yeah I thought about that too, but I can have an arbitrarily big or small percentage, eg. 235 % or -35 % so that it not really an alternative.

**Comment by bengotow** [comment]
Just out of curiosity, what happens if you make an image 32px high and 1px wide? I think it might be proportionally scaling it up, so setting the height correctly may prevent it from being that wide.

**Comment by bengotow** [comment]
I looked into what it'd take to add a "title-only" mode to the Tray API. It looks like only Mac OS X supports the "title" at all, and making the icon optional would be a little tricky. (It's required for the standard constructor, etc.)

You might want to look at what we did for N1 back when we showed the unread count in the title bar - we weren't happy with the font size and spacing used by the title attribute, so we just drew everything into a canvas and then converted the canvas into a NativeImage:

https://github.com/nylas/N1/blob/cff2f71d18268b27d71d337a3ff04c693528f12a/src/canvas-utils.coffee#L81

https://github.com/nylas/N1/blob/cff2f71d18268b27d71d337a3ff04c693528f12a/internal_packages/system-tray/lib/system-tray-icon-store.es6#L59

**Comment by MarshallOfSound** [comment]
@vikeri You can use an HTML `Canvas` element and draw arbitrary content into it.  Then you can use the canvas to generate the [Base64 PNG](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toDataURL) image code, then you can use the [Electron Native Image API](https://github.com/electron/electron/blob/master/docs/api/native-image.md#nativeimagecreatefromdataurldataurl) to convert that to an actual image you can use as your tray icon.

Just a cool workaround you can use in the meantime.

**Comment by vikeri** [comment]
Ok thanks, I'll look into those alternatives.

**Comment by mzdr** [comment]
I'm running into this issue myself right now. It's not that bad, but setting a tray without an icon would come in pretty handy.

@zcbenz Any updates on this one?

**Comment by miniak** [comment]
After https://github.com/electron/electron/pull/13851, which will be available in v2.0.9, you can create the Tray with an empty image and set the title.

**Comment by adamkdean** [comment]
For reference:

```js
tray = new Tray(nativeImage.createEmpty())
tray.setTitle('test')
```
